starknet_py/net/account/account.py

```diff
@@ -359,6 +360,57 @@ async def verify_message(self, typed_data: TypedData, signature: List[int]) -> b
         message_hash = typed_data_dataclass.message_hash(account_address=self.address)
         return await self._verify_message_hash(message_hash, signature)
 
+    @staticmethod
+    async def deploy_account(
+        *,
+        address: int,
+        class_hash: int,
+        salt: int,
+        key_pair: KeyPair,
+        chain: StarknetChainId,
+        max_fee: int,
+        client: Client,
+    ) -> "Account":
+        """
+        Deploys an account contract with provided class_hash on starknet and returns
+        a Account instance to be used.
+
+        Provided address must be first prefunded with enough tokens, otherwise the method will fail.
+
+        :param address: calculated and prefunded address of the new account.
+        :param class_hash: class_hash of the account contracat to be deployed.
+        :param salt: salt used to calculate the address.
+        :param client: a Client instance used for deployment.
+        :param key_pair: KeyPair used to calculate address and sign deploy account transaction.
+        :param chain: id of the StarkNet chain used.
+        :param max_fee: max fee to be paid for deployment, must be less or equal to the amount of tokens prefunded.
+        """
+        if address != (
+            computed := compute_address(
+                salt=salt,
+                class_hash=class_hash,  # class_hash of the Account declared on the StarkNet
+                constructor_calldata=[key_pair.public_key],
+                deployer_address=0,
+            )
+        ):
+            raise ValueError(
+                f"Provided address {hex(address)} is different than computed address {hex(computed)} "
+                f"for the given class_hash and salt"
+            )
+
+        account = Account(
+            address=address, client=client, key_pair=key_pair, chain=chain
+        )
+        deploy_account_tx = await account.sign_deploy_account_transaction(
+            class_hash=class_hash,
+            contract_address_salt=salt,
+            constructor_calldata=[key_pair.public_key],
+            max_fee=max_fee,
+        )
+        result = await client.deploy_account(deploy_account_tx)
+        await client.wait_for_tx(result.transaction_hash)
```

Comment:
I don't like waiting for txs at all. In case of mainnet it can take minutes. I suggest doing similar flow as with deploying regular contracts.

Reply:
This is a helper method to ease onboarding new accounts i.e. user creating their first account. I think they'd want to wait for their account to be usable anyway. If someone wants more flexibility they should use UDC, Contract or create deploy_account transactions themselves. This is not the point of this method.

Reply:
If users use this method in case of some network malfunction they will be left with a long wait, wondering what's going on. There will be no way of knowing the tx hash that they can check later. I suggest doing flow similar as with contracts - you get a tx that you can wait for with an additional account field.

Reply:
I get your point about the tx hash and I think there should be some way to retrieve it. However, this method is meant as kind of replacement for `AccountClient.create_account` for the less advanced users - I think that adding extra steps will increase confusion.

Reply:
I think that if documentation uses `wait_for_tx` there shouldn't be any confusion (people will just copy it to their codebases) and we'll avoid questions like "Why does it take so long to deploy account using starknet.py?".

Reply:
Okay, I'll try changing it up and we'll see if it's simple enough.